src/library/clementine/clementinedbconnection.cpp

```diff
@@ -0,0 +1,231 @@
+}
+
+QList<ClementinePlaylistEntry>
+ClementineDbConnection::getPlaylistEntries(int playlistId) const {
+    PerformanceTimer time;
+    time.start();
+
+    QList<ClementinePlaylistEntry> list;
+    ClementinePlaylistEntry entry;
+
+    //Query clementine playlist tracks
+    QSqlQuery queryPlaylist(m_database);
+    queryPlaylist.setForwardOnly(true); // Saves about 50% time
+    queryPlaylist.prepare(
+            "SELECT "
+            "ROWID, "                      // 0
+            "playlist_items.title, "       // 1
+            "playlist_items.filename, "    // 2
+            "playlist_items.length, "      // 3
+            "playlist_items.artist, "      // 4
+            "playlist_items.year, "        // 5
+            "playlist_items.album, "       // 6
+            "playlist_items.rating, "      // 7
+            "playlist_items.genre, "       // 8
+            "playlist_items.track, "       // 9
+            "playlist_items.bpm, "         // 10
+            "playlist_items.bitrate, "     // 11
+            "playlist_items.comment, "     // 12
+            "playlist_items.playcount, "   // 13
+            "playlist_items.composer, "    // 14
+            "playlist_items.grouping, "    // 15
+            "playlist_items.albumartist, " // 16
+            "playlist_items.type "         // 17
+            "FROM playlist_items "
+            "WHERE playlist_items.playlist = :playlistId");
+    queryPlaylist.bindValue(":playlistId", playlistId);
+    if (!queryPlaylist.exec()) {
+        LOG_FAILED_QUERY(queryPlaylist);
+        return {};
+    }
+
+    //Query clementine library tracks which are needed in playlist
+    //Note that we use the same indices as in the playlist query!
+    QSqlQuery queryLibrary(m_database);
+    queryLibrary.setForwardOnly(true);
+    queryLibrary.prepare(
+            "SELECT "
+            "songs.ROWID, "       // 0
+            "songs.title, "       // 1
+            "songs.filename, "    // 2
+            "songs.length, "      // 3
+            "songs.artist, "      // 4
+            "songs.year, "        // 5
+            "songs.album, "       // 6
+            "songs.rating, "      // 7
+            "songs.genre, "       // 8
+            "songs.track, "       // 9
+            "songs.bpm, "         // 10
+            "songs.bitrate, "     // 11
+            "songs.comment, "     // 12
+            "songs.playcount, "   // 13
+            "songs.composer, "    // 14
+            "songs.grouping, "    // 15
+            "songs.albumartist, " // 16
+            "songs.albumartist "  // 17
+            "FROM songs "
+            "INNER JOIN playlist_items "
+            "ON playlist_items.library_id = songs.ROWID "
+            "WHERE playlist_items.playlist = :playlistId");
+    queryLibrary.bindValue(":playlistId", playlistId);
+    if (!queryLibrary.exec()) {
+        LOG_FAILED_QUERY(queryLibrary);
+        return {};
+    }
+
+    QSqlQuery* playlistTrackDataSourceQuery = &queryPlaylist;
```

Comment:
i am not happy with that pointer over here, does anyone got a better idea than a plain pointer?

Reply:
references didnt work for me over here.